apps/req.c

```diff
@@ -1614,27 +1709,44 @@ static int do_sign_init(EVP_MD_CTX *ctx, EVP_PKEY *pkey,
         md = NULL;
     }
     if (!EVP_DigestSignInit(ctx, &pkctx, md, NULL, pkey))
```

Comment:
Is there a reason for creating another ctx here? Cant you pass in your created ctx?

Reply:
SM2 algorithm requires to construct a special value before computing the to-be-signed hash value. To fit this into current EVP framework, some special treatments have been made in #7113. To be simplified, the newly created ctx should be set into MD_CTX before `EVP_DigestSign/VerifyInit`, so it's not possible to pass it to EVP when doing `EVP_DigestSign/VerifyInit`